go/libkb/wot.go

```diff
@@ -198,48 +202,66 @@ func fetchWot(mctx MetaContext, username *string) (res []serverWotVouch, err err
 	return response.Vouches, nil
 }
 
-func FetchMyWot(mctx MetaContext) (res []keybase1.WotVouch, err error) {
-	defer mctx.Trace("FetchMyWot", func() error { return err })()
-	serverVouches, err := fetchWot(mctx, nil)
+// fetchUserWotByVoucher gets vouches written for vouchee. If voucher is
+// specified, only get the vouch (of any status) written by voucher for vouchee.
+// If voucheeUser has been previously loaded, pass it in to prevent reloading
+// (there is no harm in reloading).
+func fetchUserWotByVoucher(mctx MetaContext, vouchee string, voucher *string, voucheeUser *User) (res []keybase1.WotVouch, err error) {
+	vouches, err := fetchWot(mctx, &vouchee, voucher)
 	if err != nil {
-		mctx.Debug("error fetching pending web-of-trust vouches: %s", err.Error())
+		mctx.Debug("error fetching web-of-trust vouches for %s: %s", vouchee, err.Error())
 		return nil, err
 	}
-	me, err := LoadMe(NewLoadUserArgWithMetaContext(mctx))
-	if err != nil {
-		return nil, fmt.Errorf("error loading myself: %s", err.Error())
+	if voucheeUser == nil {
+		voucheeUser, err = LoadUser(NewLoadUserArgWithMetaContext(mctx).WithName(vouchee))
+		if err != nil {
+			return nil, fmt.Errorf("error loading vouchee: %s", err.Error())
+		}
 	}
-	for _, serverVouch := range serverVouches {
-		vouch, err := transformUserVouch(mctx, serverVouch, me)
+	for _, serverVouch := range vouches {
+		vouch, err := transformUserVouch(mctx, serverVouch, voucheeUser)
 		if err != nil {
-			mctx.Debug("error validating server-reported pending web-of-trust vouches: %s", err.Error())
+			mctx.Debug("error validating server-reported web-of-trust vouches for %s: %s", vouchee, err.Error())
 			return nil, err
 		}
 		res = append(res, vouch)
 	}
-	mctx.Debug("found %d web-of-trust vouches", len(res))
+	mctx.Debug("found %d web-of-trust vouches for %s", vouchee, len(res))
 	return res, nil
 }
 
-func FetchUserWot(mctx MetaContext, username string) (res []keybase1.WotVouch, err error) {
-	defer mctx.Trace("FetchUserWot", func() error { return err })()
-	vouches, err := fetchWot(mctx, &username)
+// FetchMyWot gets all vouches (of any status) written for me
+func FetchMyWot(mctx MetaContext) (res []keybase1.WotVouch, err error) {
+	defer mctx.Trace("FetchMyWot", func() error { return err })()
+	me, err := LoadMe(NewLoadUserArgWithMetaContext(mctx))
 	if err != nil {
-		mctx.Debug("error fetching web-of-trust vouches for %s: %s", username, err.Error())
-		return nil, err
+		return nil, fmt.Errorf("error loading myself: %s", err.Error())
 	}
-	vouchee, err := LoadUser(NewLoadUserArgWithMetaContext(mctx).WithName(username))
+	myName := me.GetName()
+	return fetchUserWotByVoucher(mctx, myName, nil, me)
```

Comment:
(vouchee=me, voucher=nil) and (vouchee=nil, voucher=nil) return the same results (all vouches written for me), but explicitly setting vouchee=me is easier to read and more consistent with `FetchMyWotByUser`'s implementation. either way the LoadMe/LoadUser call needs to be done at some point in the method call.